src/core/auth.ts

```diff
@@ -47,33 +43,28 @@ export async function validateAuthorizationIntegrity(
     throw new Error('expected no more than 1 signature for authorization');
   }
 
+  // validate payload integrity
   const payloadJson = Jws.decodePlainObjectPayload(message.authorization);
-  const { descriptorCid } = payloadJson;
+
+  const dwnInterface = message.descriptor.interface;
+  const dwnMethod = message.descriptor.method;
+
+  let schemaLookupKey: string;
+  if (dwnInterface === DwnInterfaceName.Records && dwnMethod === DwnMethodName.Write) {
+    schemaLookupKey = 'RecordsWriteDecodedAuthorizationPayload';
+  } else {
+    schemaLookupKey = 'BaseDecodedAuthorizationPayload';
+  }
+  validateJsonSchema(schemaLookupKey, payloadJson);
+
```

Comment:
Update from a conversation we had offline:

* JSON Schema are worth keeping because they are documentation-like and they validate the shape of different auth payloads
* It doesn't make sense for `validateAuthorizationIntegrity` to check that all properties other than `descriptorCid` are CIDs. We'll leave that responsibility up to each DWN interface (currently, only `RecordsWrite.validateIntegrity`).
* We don't want this function to be DWN interface dependence, so as a middle-ground, we are passing the JSON schema key as a parameter.